EventListener/TokenSubscriber.php

```diff
@@ -323,4 +333,138 @@ private function getCustomFieldValues(CustomObject $customObject, Token $token,
+                        (int) explode('cmf_', $condition['field'])[1]
+                    );
+                    $customObject = $customField->getCustomObject();
+                    $fieldAlias   = $customField->getAlias();
+                } elseif ('cmo_' === substr($condition['field'], 0, 4)) {
+                    $customObject = $this->customObjectModel->fetchEntity(
+                        (int) explode('cmo_', $condition['field'])[1]
+                    );
+                    $fieldAlias   = 'name';
+                } else {
+                    continue;
+                }
+
+                $key = $customObject->getId().'-'.$leadId;
+                if (!isset($cachedCustomItems[$key])) {
+                    $cachedCustomItems[$key] = $this->getCustomItems($customObject, $leadId);
+                }
+
+                $result = $this->getCustomFieldValue($customObject, $fieldAlias, $cachedCustomItems[$key]);
+
+                $customFieldValues[$condition['field']] = $result;
+            } catch (NotFoundException|InvalidCustomObjectFormatListException $e) {
+                continue;
+            }
+        }
+
+        return $customFieldValues;
+    }
+
+    /**
+     * @param array<mixed> $customItems
+     *
+     * @throws InvalidCustomObjectFormatListException
+     */
+    private function getCustomFieldValue(CustomObject $customObject, string $customFieldAlias, array $customItems): string
+    {
+        $fieldValues = [];
+
+        foreach ($customItems as $customItemData) {
+            // Name is known from the CI data array.
+            if ('name' === $customFieldAlias) {
+                $fieldValues[] = $customItemData['name'];
+
+                continue;
+            }
+
+            // Custom Field values are handled like this.
+            $customItem = new CustomItem($customObject);
+            $customItem->populateFromArray($customItemData);
+            $customItem = $this->customItemModel->populateCustomFields($customItem);
+
+            try {
+                $fieldValue = $customItem->findCustomFieldValueForFieldAlias($customFieldAlias);
+                // If the CO item doesn't have a value, get the default value
+                if (empty($fieldValue->getValue())) {
+                    $fieldValue->setValue($fieldValue->getCustomField()->getDefaultValue());
+                }
+                $fieldValues[] = $fieldValue->getCustomField()->getTypeObject()->valueToString($fieldValue);
+            } catch (NotFoundException $e) {
+                // Custom field not found.
+            }
+        }
+
+        return $this->tokenFormatter->format($fieldValues, TokenFormatter::DEFAULT_FORMAT);
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    private function getCustomItems(CustomObject $customObject, int $leadId): array
+    {
+        $orderBy  = CustomItem::TABLE_ALIAS.'.id';
+        $orderDir = 'DESC';
+
+        $tableConfig = new TableConfig(1, 1, $orderBy, $orderDir);
```

Comment:
The method is called `getCustomItems()` but it will always get only 1, right? As the limit = 1 here. What if there is more custom items linked to the contact?